tools/Customization/DevHome.FileExplorerSourceControlIntegration/Services/RepositoryTracking.cs

```diff
@@ -134,10 +141,11 @@ public string GetSourceControlProviderForRootPath(string rootPath)
     {
         lock (trackRepoLock)
         {
-            if (TrackedRepositories.TryGetValue(rootPath, out var value))
+            var rootPathRegistered = TrackedRepositories.Keys.FirstOrDefault(key => StringComparer.OrdinalIgnoreCase.Equals(key, rootPath));

```

Comment:
I believe this can be simplified to `if(TrackRepositories.TryGetValue(key, trackedRepo) { return trackedRepo;}`

Reply:
As the search is case insensitive, TryGetValue cannot be used here. Thanks!

Reply:
Using a custom lookup, you'll still be able to insert duplicate paths that differ only in case. (HINT: This would be a good test case to add and validate).

Instead, make the entire Dictionary enforce the desired comparisons automatically by overriding the comparer. According to https://learn.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings?redirectedfrom=MSDN#choosing-a-stringcomparison-member-for-your-method-call, the appropriate comparer for file paths is [StringComparer.OrdinalIgnoreCase](https://learn.microsoft.com/en-us/dotnet/api/system.stringcomparer.ordinalignorecase?view=net-8.0).

So when you create your dictionary, do
```cs
TrackedRepositories = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
```

Reply:
Resolved in latest. Thank you!!